### PR TITLE
setCustomLocking will now check for descendant content objects

### PR DESCRIPTION
**Pull request by jamesrea83**
in reference to #2469

### CODE REVIEW

**Comment by oliverfoster** [comment]
Original function just checks the immediate children using `this.shouldLock` (which is a help that doesn't care about its own instance).
```js
        setCustomLocking: function() {
            var children = this.getAvailableChildModels();

            for (var i = 0, j = children.length; i < j; i++) {
                var child = children[i];

                child.set("_isLocked", this.shouldLock(child));
            }
        },
```
You're recursively calling the `this.setCustomLocking` function on this parent instance with the immediate children's children.
```js
        setCustomLocking: function(childModels) {
            var children = childModels || this.getAvailableChildModels();

            for (var i = 0, j = children.length; i < j; i++) {
                var child = children[i];

                child.set("_isLocked", this.shouldLock(child));

                if (child.get('_type') === 'menu') this.setCustomLocking(child.getAvailableChildModels());
            }
        },
```
Would not calling it on the immediate child work just as well as `this.shouldLock` is instance agnostic?
```js
        setCustomLocking: function() {
            var children = this.getAvailableChildModels();

            for (var i = 0, j = children.length; i < j; i++) {
                var child = children[i];

                child.set("_isLocked", this.shouldLock(child));

                if (child.get('_type') === 'menu') child.setCustomLocking();
            }
        },
```

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "adapt_framework",
-    "version": "4.2.0",
+    "version": "4.2.1",
```
Comment (oliverfoster) [inline package.json]: we'd normally do this when the fw is released.

File: src/core/js/models/adaptModel.js [hunk 2]
```diff
@@ -719,6 +719,8 @@ define([
                 var child = children[i];
 
                 child.set("_isLocked", this.shouldLock(child));
+
+                if (child.get('_type') === 'menu') child.setCustomLocking();
```
Comment (oliverfoster) [inline src/core/js/models/adaptModel.js]: I've had another thought on this today (apologies).

### Assumption
Works for only course or menu contentObjects with child menus.
The children will have custom locking if the parent has custom locking.

### Issue
The child menu might have an alternative locking type.

### Suggestion
Switch from `child.setCustomLocking();` to `child.checkLocking();`

```suggestion
                if (child.get('_type') === 'menu') {
                    child.checkLocking();
                }
```

**Comment by moloko** [comment]
@jamesrea83 you have a conflict I'm afraid...

**Pull request merged**